apps/andi/lib/andi/input_schemas/datasets/technical.ex

```diff
@@ -87,11 +87,17 @@ defmodule Andi.InputSchemas.Datasets.Technical do
 
   def preload(struct), do: StructTools.preload(struct, [:schema, :sourceQueryParams, :sourceHeaders])
 
-  defp validate_top_level_selector(%{changes: %{sourceFormat: source_format}} = changeset)
-       when source_format in ["xml", "text/xml"] do
+  defp validate_top_level_selector(%{changes: %{sourceFormat: source_format}} = changeset) when source_format in ["xml", "text/xml"] do
     validate_required(changeset, [:topLevelSelector], message: "is required")
   end
 
+  defp validate_top_level_selector(%{changes: %{sourceFormat: "application/json", topLevelSelector: top_level_selector}} = changeset) do
```

Comment:
is there a reason this doesn't include the "json" option similar to the "xml" one?

Reply:
nope